manifests/allow_client_on_server.pp

```diff
@@ -0,0 +1,56 @@
+# @summary Enable backup client to access backup server
+#
+# @param hostname
+#   fqdn hostname of backup client.
+#
+# @param ip
+#   ip of backup client.
+#
+# @param key
+#   The public ssh key itself; generally a long string...
+#
+# @param key_type
+#   The encryption type used in the public ssh key.
+#
+# @example
+#   profile_backup::allow_client_on_server { 'allow host hostname access to backup servers':
+#     'hostname' => String,
+#     'ip' => String,
+#     'key' => String,
+#     'key_type' => String,
+#   }
+define profile_backup::allow_client_on_server (
+  String $hostname,
+  String $ip,
+  String $key,
+  String $key_type,
+) {
+  # SET DEFAULTS FOR CLIENT
+  $repo_path = "${profile_backup::server::backup_directory}/${hostname}"
```

Comment:
$profile_backup::server::backup_directory might need a manual hiera lookup()

Reply:
It works as is.